autogluon/utils/tabular/ml/trainer/abstract_trainer.py

```diff
@@ -128,6 +127,7 @@ def __init__(self, path: str, problem_type: str, scheduler_options=None, objecti
         self.num_cols_train = None
 
         self.is_data_saved = False
+        self.normalize_predprobs = False # whether or not probabilistic predictions may need to be renormalized (eg. distillation with BINARY -> REGRESSION)
```

Comment:
Shouldn't this live in the models? Whether we normalize depends on which model we are getting results from, this could be a distilled model or a normal model, or both.

Reply:
This is like a final safety check that one may sometimes wish to apply (eg. when changing problem_types as in distillation).  I agree all individual models should be outputting normalized probabilities in the first place.  But example in distillation, the individual models currently think they are doing a regression task, and thus it might be weird to ask them to normalize in this case.  I think have this (optional) safety check for the final overall prediction may be useful in future applications as well.

Reply:
Ok, although can you add a TODO above this that mentions long-term we should look to have each model do its own normalization, and Trainer can specify to normalize each model by checking the model's problem_type upon loading and calling predict_proba()